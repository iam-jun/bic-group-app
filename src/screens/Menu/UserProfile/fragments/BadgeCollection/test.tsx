import React from 'react';
import {
  View, Dimensions, FlatList, StyleSheet,
} from 'react-native';
import { ExtendedTheme, useTheme } from '@react-navigation/native';
import Avatar from '~/baseComponents/Avatar';
import Icon from '~/baseComponents/Icon';
import spacing, { borderRadius } from '~/theme/spacing';

const Grid = ({ data }: {data: any[]}) => {
  const deviceWidth = Dimensions.get('window').width;

  const itemWidth = 48; // Constant item width
  const numColumns = Math.floor((deviceWidth - 10 * 2) / (itemWidth + 6 * 2));
  const rowWidth = numColumns * (itemWidth + 6 * 2);
  const paddingLeft = (deviceWidth - (10 * 2) - rowWidth) / 2;

  const theme: ExtendedTheme = useTheme();
  const { colors } = theme;
  const styles = themeStyles(theme);

  const isSelected = false;

  const renderGridItem = ({ item, index }: any) =>
    // Render each grid item here
    (
      <View
        key={index}
        style={{
          margin: 6,
          borderWidth: isSelected ? 2 : 1,
          borderColor: isSelected ? colors.purple50 : colors.neutral5,
          borderRadius: borderRadius.pill,
        }}
      >
        <Avatar.Medium
          isRounded
          source={{ uri: item?.iconUrl }}
        />
        {isSelected
          ? (
            <View style={styles.iconChose}>
              <Icon size={10} icon="Check" tintColor={colors.white} />
            </View>
          )
          : null}
      </View>
    );

  return (
    <View style={{
      flex: 1,
      justifyContent: 'center',
      alignItems: 'center',
      paddingHorizontal: 10,
      paddingBottom: 10,
    }}
    >

      <View style={{
        // paddingLeft,
      }}
      >
        <FlatList
          data={data}
          renderItem={renderGridItem}
          keyExtractor={(item, index) => index.toString()}
          numColumns={numColumns}
        />
      </View>
    </View>
  );
};

const themeStyles = (theme: ExtendedTheme) => {
  const { colors } = theme;

  return StyleSheet.create({
    container: {
      flex: 1,
      paddingTop: spacing.padding.large,
    },
    iconChose: {
      backgroundColor: colors.purple50,
      borderRadius: borderRadius.pill,
      height: 20,
      width: 20,
      alignItems: 'center',
      justifyContent: 'center',
      position: 'absolute',
      zIndex: 1,
      right: 0,
    },
  });
};

export default Grid;
