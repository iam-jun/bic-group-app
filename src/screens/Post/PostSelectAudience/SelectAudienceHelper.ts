import { isEqual } from 'lodash';
import { ICreatePostParams } from '~/interfaces/IPost';

export interface ISelectAudienceParams extends ICreatePostParams {
  isFirstStep?: boolean;
}

export const checkChangeAudiences = (
  a1: any, a2: any,
) => {
  if (a1?.length !== a2?.length) {
    return true;
  }
  const compare = (
    x: any, y: any,
  ) => (x > y ? 1 : -1);
  const ids1: any[] = [];
  const ids2: any[] = [];
  a1?.map?.((a: any) => ids1.push(a?.id));
  a2?.map?.((a: any) => ids2.push(a?.id));
  return !isEqual(
    ids1.sort(compare), ids2.sort(compare),
  );
};
