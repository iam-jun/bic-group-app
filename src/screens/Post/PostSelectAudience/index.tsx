import React, {useState, useEffect} from 'react';
import {View, StyleSheet, SectionList} from 'react-native';
import {useTheme} from 'react-native-paper';
import {useDispatch} from 'react-redux';
import debounce from 'lodash/debounce';

import {useBaseHook} from '~/hooks';
import {useCreatePost} from '~/hooks/post';
import {ITheme} from '~/theme/interfaces';
import postActions from '~/screens/Post/redux/actions';

import ScreenWrapper from '~/beinComponents/ScreenWrapper';
import Header from '~/beinComponents/Header';
import SearchInput from '~/beinComponents/inputs/SearchInput';
import Text from '~/beinComponents/Text';
import PrimaryItem from '~/beinComponents/list/items/PrimaryItem';
import SelectingAudiences from '~/screens/Post/components/SelectingAudiences';

import postDataHelper from '~/screens/Post/helper/PostDataHelper';
import {IGroup} from '~/interfaces/IGroup';
import {OnChangeCheckedGroupsData} from '~/beinComponents/GroupTree';
import FlatGroupItem from '~/beinComponents/list/items/FlatGroupItem';
import {useRootNavigation} from '~/hooks/navigation';
import {IUser} from '~/interfaces/IAuth';

const PostSelectAudience = () => {
  const [selectingAudiences, setSelectingAudiences] = useState<
    (IGroup | IUser)[]
  >([]);
  const [selectingGroups, setSelectingGroups] = useState<{[x: string]: IGroup}>(
    {},
  );
  const [selectingUsers, setSelectingUsers] = useState<{[x: string]: IUser}>(
    {},
  );

  const dispatch = useDispatch();
  const {t} = useBaseHook();
  const {rootNavigation} = useRootNavigation();
  const theme: ITheme = useTheme() as ITheme;
  const {spacing} = theme;
  const styles = createStyle(theme);

  const createPostData = useCreatePost();
  const {
    chosenAudiences,
    searchResultAudienceGroups,
    searchResultAudienceUsers,
  } = createPostData || {};

  const sectionListData: any = [];
  if (searchResultAudienceGroups?.length > 0) {
    sectionListData.push({
      title: t('post:label_groups'),
      data: searchResultAudienceGroups,
    });
  }
  if (searchResultAudienceUsers?.length > 0) {
    sectionListData.push({
      title: t('post:label_users'),
      data: searchResultAudienceUsers,
    });
  }

  const updateSelectingAudiences = () => {
    const newSelectingAudiences: (IUser | IGroup)[] = [];
    Object.values(selectingGroups).map((group: IGroup) => {
      if (group) {
        newSelectingAudiences.push(group);
      }
    });
    Object.values(selectingUsers).map((user: IUser) => {
      if (user) {
        newSelectingAudiences.push(user);
      }
    });
    setSelectingAudiences(newSelectingAudiences);
  };

  useEffect(() => {
    if (selectingAudiences?.length === 0) {
      const newSelectingUsers: any = {};
      const newSelectingGroups: any = {};
      chosenAudiences?.map?.((item: any) => {
        if (item && item?.type === 'user') {
          newSelectingUsers[item.id] = item;
        } else {
          newSelectingGroups[item.id] = item;
        }
      });
      setSelectingUsers(newSelectingUsers);
      setSelectingGroups(newSelectingGroups);
    }
  }, [chosenAudiences]);

  useEffect(() => {
    updateSelectingAudiences();
  }, [selectingGroups]);

  useEffect(() => {
    updateSelectingAudiences();
  }, [selectingUsers]);

  const onPressSave = () => {
    dispatch(postActions.setCreatePostChosenAudiences(selectingAudiences));
    rootNavigation.goBack();
  };

  const onRemoveItem = (item: any) => {
    if (item.type === 'user') {
      const newSelectingUsers: any = {...selectingUsers};
      newSelectingUsers[item.id] = false;
      setSelectingUsers(newSelectingUsers);
    } else {
      const newSelectingGroups: any = {...selectingGroups};
      newSelectingGroups[item.id] = false;
      setSelectingGroups(newSelectingGroups);
    }
  };

  const onChangeCheckedGroups = (data: OnChangeCheckedGroupsData) => {
    setSelectingGroups({...selectingGroups, ...data} as any);
  };

  const getSmallestChild = (
    smallestGroup: IGroup,
    newGroups: IGroup[],
    treeData: IGroup,
  ): any => {
    if (smallestGroup?.children?.[0]) {
      return getSmallestChild(
        smallestGroup?.children?.[0],
        newGroups,
        treeData,
      );
    } else {
      newGroups.push({...smallestGroup, treeData: treeData});
    }
  };

  const onSearch = debounce((searchText: string) => {
    postDataHelper
      .getSearchAudiences(searchText)
      .then(response => {
        if (response && response.data) {
          const {users = [], groups = []} = response?.data || {};

          dispatch(postActions.setSearchResultAudienceGroups(groups));

          const newListUsers: any = [];
          users?.map?.((item: any) => {
            newListUsers.push({
              id: item.id,
              type: 'user',
              name: item.fullname || item.username,
              avatar: item.avatar,
            });
          });
          dispatch(postActions.setSearchResultAudienceUsers(newListUsers));
        }
      })
      .catch(e => {
        console.log('\x1b[31m', '🐣️ getSearchAudiences |  : ', e, '\x1b[0m');
      });
  }, 500);

  const onChangeTextSearch = (text: string) => {
    onSearch(text);
  };

  const onPressUser = (user: any) => {
    const newSelectingUsers: any = {...selectingUsers};
    if (newSelectingUsers[user.id]) {
      newSelectingUsers[user.id] = false;
    } else {
      newSelectingUsers[user.id] = user;
    }
    setSelectingUsers(newSelectingUsers);
  };

  const renderItem = ({item}: any) => {
    const {id, name, icon, avatar, type} = item || {};
    const isGroup = type !== 'user';

    if (isGroup) {
      return (
        <FlatGroupItem
          {...item}
          initShowTree={false}
          hidePath={false}
          selectingData={selectingGroups}
          showSmallestChild
          onChangeCheckedGroups={onChangeCheckedGroups}
        />
      );
    } else {
      return (
        <PrimaryItem
          showAvatar
          avatar={icon || avatar}
          avatarProps={{variant: isGroup ? 'ultraLarge' : 'large'}}
          style={styles.item}
          title={name}
          onPressCheckbox={() => onPressUser(item)}
          onPress={() => onPressUser(item)}
          checkboxProps={{
            style: {position: 'absolute', left: 26, bottom: 0},
            isChecked: !!selectingUsers[id],
          }}
        />
      );
    }
  };

  const renderListHeader = () => {
    if (
      searchResultAudienceGroups?.length === 0 &&
      searchResultAudienceUsers?.length === 0
    ) {
      return null;
    }
    return (
      <Text.H5 style={{marginVertical: spacing?.margin.small}}>
        Search Results
      </Text.H5>
    );
  };

  const renderSectionHeader = ({section: {title}}: any) => {
    return <Text.H6 style={styles.sectionHeader}>{title}</Text.H6>;
  };

  return (
    <ScreenWrapper style={styles.container}>
      <Header
        title={'post:select_audience'}
        titleTextProps={{useI18n: true}}
        buttonText={'post:save'}
        buttonProps={{useI18n: true}}
        onPressButton={onPressSave}
      />
      <SearchInput
        style={styles.searchInput}
        onChangeText={onChangeTextSearch}
        placeholder={t('post:search_audiences_placeholder')}
      />
      <SelectingAudiences
        list={selectingAudiences}
        onRemoveItem={onRemoveItem}
      />
      <SectionList
        style={{paddingHorizontal: spacing?.padding.large}}
        sections={sectionListData}
        keyExtractor={(item, index) => `section_list_${item}_${index}`}
        ListHeaderComponent={renderListHeader}
        renderSectionHeader={renderSectionHeader}
        renderItem={renderItem}
        ItemSeparatorComponent={() => (
          <View style={{height: spacing?.margin.large}} />
        )}
      />
    </ScreenWrapper>
  );
};

const createStyle = (theme: ITheme) => {
  const {spacing, colors} = theme;
  return StyleSheet.create({
    container: {},
    searchInput: {
      margin: spacing?.margin.base,
    },
    item: {
      height: undefined,
      paddingHorizontal: undefined,
    },
    sectionHeader: {
      marginVertical: spacing?.margin.small,
      backgroundColor: colors.background,
    },
  });
};

export default PostSelectAudience;
