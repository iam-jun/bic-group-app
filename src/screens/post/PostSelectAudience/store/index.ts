import { createZustand, resetZustand } from '~/store/utils';
import getAudienceTree from './actions/getAudienceTree';
import ISelectAudienceState from '~/screens/post/PostSelectAudience/store/Interface';
import getAudienceSearch from './actions/getAudienceSearch';

const initState: ISelectAudienceState = {
  tree: {
    data: undefined,
    loading: true,
  },
  search: {
    key: '',
    data: undefined,
    loading: true,
  },
};

const selectAudienceStore = (set, get) => ({
  ...initState,

  setSearch: (payload?: any) => {
    set((state) => {
      state.search = payload || {};
    }, false, 'setSearch');
  },

  dispatchGetAudienceTree: getAudienceTree(set, get),
  dispatchGetAudienceSearch: getAudienceSearch(set, get),

  reset: () => resetZustand(initState, set),
});

const useSelectAudienceStore = createZustand<ISelectAudienceState>(
  'select-audience-store', selectAudienceStore,
);

export default useSelectAudienceStore;
