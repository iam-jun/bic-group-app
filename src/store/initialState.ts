import {appInitState} from '~/store/app/reducer';
import {modalInitState} from '~/store/modal/reducer';
import {authInitState} from '~/screens/Auth/redux/reducer';
import {postInitState} from '~/screens/Post/redux/reducer';
import {groupInitState} from '~/screens/Groups/redux/reducer';
import {homeInitState} from '~/screens/Home/redux/reducer';
import {notiInitState} from '~/screens/Notification/redux/reducer';
import {menuInitState} from '~/screens/Menu/redux/reducer';
import {noInternetInitState} from '~/screens/NoInternet/redux/reducer';
import {mentionInputInitState} from '~/beinComponents/inputs/MentionInput/redux/reducer';

const initialState = {
  app: appInitState,
  modal: modalInitState,
  auth: authInitState,
  post: postInitState,
  groups: groupInitState,
  home: homeInitState,
  notifications: notiInitState,
  menu: menuInitState,
  noInternet: noInternetInitState,
  mentionInput: mentionInputInitState,
};

export default initialState;
