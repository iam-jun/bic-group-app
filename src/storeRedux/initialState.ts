import { appInitState } from '~/storeRedux/app/reducer';
import { modalInitState } from '~/storeRedux/modal/reducer';
import { authInitState } from '~/storeRedux/auth/reducer';
import { postInitState } from '~/storeRedux/post/reducer';
import { groupInitState } from '~/storeRedux/groups/reducer';
import { homeInitState } from '~/storeRedux/home/reducer';
import { notiInitState } from '~/storeRedux/notification/reducer';
import { menuInitState } from '~/storeRedux/menu/reducer';
import { noInternetInitState } from '~/storeRedux/network/reducer';
import { mentionInputInitState } from '~/beinComponents/inputs/MentionInput/redux/reducer';
import { giphyInitState } from './giphy/reducer';

const initialState = {
  app: appInitState,
  modal: modalInitState,
  auth: authInitState,
  post: postInitState,
  groups: groupInitState,
  home: homeInitState,
  notifications: notiInitState,
  menu: menuInitState,
  noInternet: noInternetInitState,
  mentionInput: mentionInputInitState,
  giphy: giphyInitState,
};

export default initialState;
