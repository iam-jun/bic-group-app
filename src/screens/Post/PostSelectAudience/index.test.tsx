import React from 'react';

import {configureStore, renderWithRedux} from '~/test/testUtils';
import initialState from '~/store/initialState';
import PostSelectAudience from '.';

describe('PostSelectAudience screen', () => {
  it(`renders correctly`, async () => {
    const wrapper = renderWithRedux(<PostSelectAudience />);
    const rendered = wrapper.toJSON();
    expect(rendered).toMatchSnapshot();
  });
});
