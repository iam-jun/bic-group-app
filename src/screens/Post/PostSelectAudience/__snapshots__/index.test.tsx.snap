// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostSelectAudience screen renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "flex": 1,
    }
  }
>
  <View
    testID="header"
  >
    <View
      animatedStyle={
        Object {
          "value": Object {
            "height": 60,
          },
        }
      }
      collapsable={false}
      style={
        Object {
          "alignItems": "flex-end",
          "backgroundColor": "#FFFFFF",
          "borderBottomWidth": 0.5,
          "borderColor": "#EAEDF2",
          "elevation": 2,
          "flexDirection": "row",
          "height": 60,
          "overflow": "hidden",
          "paddingTop": 0,
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 1,
            "width": 0,
          },
          "shadowOpacity": 0.1,
          "shadowRadius": 1,
        }
      }
      testID="header.content"
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#FFFFFF",
            "flex": 1,
            "flexDirection": "row",
            "height": 60,
            "overflow": "hidden",
            "paddingLeft": 8,
            "paddingRight": 8,
          }
        }
      >
        <View
          accessibilityState={
            Object {
              "disabled": false,
            }
          }
          accessible={true}
          focusable={true}
          hitSlop={
            Object {
              "bottom": 20,
              "left": 20,
              "right": 20,
              "top": 20,
            }
          }
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID="header.back.button"
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                },
                Object {
                  "alignItems": "center",
                  "height": 48,
                  "justifyContent": "center",
                  "padding": 12,
                  "width": 48,
                },
                Object {
                  "backgroundColor": undefined,
                },
              ]
            }
          >
            <View
              style={
                Array [
                  undefined,
                  undefined,
                  undefined,
                ]
              }
            >
              <View
                className="icon-wrapper"
                testID="header.back"
              >
                <RNSVGSvgView
                  align="xMidYMid"
                  bbHeight={24}
                  bbWidth={24}
                  fill="#2F3945"
                  focusable={false}
                  height={24}
                  meetOrSlice={0}
                  minX={0}
                  minY={0}
                  style={
                    Array [
                      Object {
                        "backgroundColor": "transparent",
                        "borderWidth": 0,
                      },
                      Object {},
                      Object {
                        "flex": 0,
                        "height": 24,
                        "width": 24,
                      },
                    ]
                  }
                  vbHeight={24}
                  vbWidth={24}
                  width={24}
                >
                  <RNSVGGroup
                    fill={4281284933}
                    propList={
                      Array [
                        "fill",
                      ]
                    }
                  >
                    <RNSVGPath
                      d="M8.5,12.8l5.7,5.6c0.4,0.4,1,0.4,1.4,0c0,0,0,0,0,0c0.4-0.4,0.4-1,0-1.4l-4.9-5l4.9-5c0.4-0.4,0.4-1,0-1.4c-0.2-0.2-0.4-0.3-0.7-0.3c-0.3,0-0.5,0.1-0.7,0.3l-5.7,5.6C8.1,11.7,8.1,12.3,8.5,12.8C8.5,12.7,8.5,12.7,8.5,12.8z"
                    />
                  </RNSVGGroup>
                </RNSVGSvgView>
              </View>
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "flex": 1,
              "height": "100%",
              "justifyContent": "center",
              "marginLeft": 16,
              "paddingTop": 1.5,
            }
          }
        >
          <View
            accessibilityState={
              Object {
                "disabled": true,
              }
            }
            accessible={true}
            focusable={false}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <Text
              numberOfLines={1}
              style={
                Object {
                  "color": "#2F3945",
                  "fontFamily": "OpenSans-SemiBold",
                  "fontSize": 16,
                  "height": 24,
                  "lineHeight": 24,
                }
              }
              testID="header.text"
            >
              Select Audiences
            </Text>
          </View>
        </View>
        <View
          accessibilityState={
            Object {
              "disabled": true,
            }
          }
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#D5DCE5",
              "borderColor": "#7843DD",
              "borderRadius": 6,
              "borderWidth": 0,
              "height": 40,
              "marginRight": 4,
              "padding": 8,
            }
          }
          testID="select_audience.btn_done"
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "center",
              }
            }
            testID="button_wrapper.content"
          >
            <Text
              style={
                Object {
                  "color": "#B2BDCD",
                  "fontFamily": "OpenSans-SemiBold",
                  "fontSize": 16,
                  "textAlign": "center",
                }
              }
              testID="header.button.text"
            >
              Done
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "#FFFFFF",
            "height": 0,
            "position": "absolute",
            "top": 0,
            "width": "100%",
          }
        }
      />
    </View>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#EAEDF2",
        "borderRadius": 20,
        "height": 40,
        "justifyContent": "center",
        "margin": 12,
        "paddingHorizontal": 16,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
    >
      <View
        accessibilityState={
          Object {
            "disabled": true,
          }
        }
        accessible={true}
        focusable={false}
        hitSlop={
          Object {
            "bottom": 10,
            "left": 10,
            "right": 10,
            "top": 10,
          }
        }
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "flexDirection": "row",
              },
              Object {
                "marginRight": 8,
              },
              Object {
                "backgroundColor": undefined,
              },
            ]
          }
        >
          <View
            style={
              Array [
                undefined,
                undefined,
                undefined,
              ]
            }
          >
            <View
              className="icon-wrapper"
            >
              <RNSVGSvgView
                align="xMidYMid"
                bbHeight={20}
                bbWidth={20}
                fill="#627286"
                focusable={false}
                height={20}
                meetOrSlice={0}
                minX={0}
                minY={0}
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                      "borderWidth": 0,
                    },
                    Object {},
                    Object {
                      "flex": 0,
                      "height": 20,
                      "width": 20,
                    },
                  ]
                }
                vbHeight={24}
                vbWidth={24}
                width={20}
              >
                <RNSVGGroup
                  fill={4284641926}
                  propList={
                    Array [
                      "fill",
                    ]
                  }
                >
                  <RNSVGPath
                    d="M21.71,20.29,18,16.61A9,9,0,1,0,16.61,18l3.68,3.68a1,1,0,0,0,1.42,0A1,1,0,0,0,21.71,20.29ZM11,18a7,7,0,1,1,7-7A7,7,0,0,1,11,18Z"
                  />
                </RNSVGGroup>
              </RNSVGSvgView>
            </View>
          </View>
        </View>
      </View>
      <TextInput
        autoComplete="off"
        autoFocus={true}
        onChangeText={[Function]}
        onFocus={[Function]}
        onSubmitEditing={[Function]}
        placeholder="Search for groups or people"
        placeholderTextColor="#627286"
        returnKeyType="search"
        selectionColor="#627286"
        style={
          Object {
            "color": "#2F3945",
            "flex": 1,
            "fontFamily": "OpenSans-Regular",
            "fontSize": 16,
            "height": "100%",
          }
        }
        testID="post_select_audience.search"
        value=""
      />
    </View>
  </View>
  <RCTScrollView
    ListEmptyComponent={[Function]}
    ListFooterComponent={[Function]}
    ListHeaderComponent={[Function]}
    data={Array []}
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    style={
      Object {
        "paddingHorizontal": 16,
      }
    }
  >
    <View>
      <View
        onLayout={[Function]}
      />
      <View
        onLayout={[Function]}
      >
        <View>
          <ActivityIndicator
            color="#EAEDF2"
            size="large"
          />
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
